fluentd/out_syslog.rb

```diff
@@ -81,17 +97,31 @@ def emit(tag, es, chain)
           time = Time.now
         end
         @packet.time = time
-        @packet.tag = if @tag_key
-                        begin
-                          record[@tag_key][0..31].gsub(/[\[\]\s]/,'') # tag is trimmed to 32 chars for syslog_protocol gem compatibility
-                        rescue
-                          tag[0..31] # tag is trimmed to 32 chars for syslog_protocol gem compatibility
-                        end
-                      else
-                          tag[0..31] # tag is trimmed to 32 chars for syslog_protocol gem compatibility
-                      end
+        if @tag_key.any?
+          @tag_key.each { |tkey|
+            mytag = record
+            tkey.split('.').each { |p|
```

Comment:
This could be more elegantly expressed using #keep_if or #select

https://ruby-doc.org/core-2.2.0/Array.html#method-i-keep_if